tinyinst.cpp

```diff
@@ -1125,9 +1125,20 @@ void TinyInst::Init(int argc, char **argv) {
 
   std::list <char *> module_names;
   GetOptionAll("-instrument_module", argc, argv, &module_names);
-  for (auto iter = module_names.begin(); iter != module_names.end(); iter++) {
+#if defined __APPLE__ && defined ARM64

```

Comment:
I think this functionality should be configurable via a flag (-page_extend_modules or something like that). It can be enabled by default on ARM, but I think it would be useful to be able to disable it. Specifically, there might be cases where the code on overlapping pages isn't used. In this cases, only instrumenting the intended library might give better performance.

Additionally, I think this mechanism of extending the list of instrumented modules should be documented in https://github.com/googleprojectzero/TinyInst/blob/master/macOS/README.md

Reply:
added flag